### DISCUSSION TITLE
help haugene/transmission-openvpn ivacy

### DISCUSSION THREAD

**Discussion by caladol1** [haugene/docker-transmission-openvpn#2023] [Q&A] [answered]
Hi all I am sort of new to docker I setup portainer in a synology nas and I am trying to deploy haugene/transmission-openvpn
my default.ovpn
```
client
dev tun
proto udp
remote vlus-pa2-ovpn-tcp.dns2use.com 80
persist-key
persist-tun
ca /etc/openvpn/ivacy/ca.crt
tls-auth /etc/openvpn/ivacy/Wdc.key 1
cipher AES-256-CBC
comp-lzo
verb 1
mute 20
route-method exe
route-delay 2
route 0.0.0.0 0.0.0.0
float
auth-user-pass /etc/openvpn/ivacy/openvpn-credentials.txt
```

my composer code

version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
         - '/volume2/docker/transmition/VPN/default.ovpn:/etc/openvpn/ivacy/default.ovpn'
         - '/volume2/docker/transmition/VPN/Wdc.key:/etc/openvpn/ivacy/Wdc.key'
         - '/volume2/docker/transmition/VPN/ca.crt:/etc/openvpn/ivacy/ca.crt'
         - '/volume2/docker/transmition/VPN/openvpn-credentials.txt:/etc/openvpn/ivacy/openvpn-credentials.txt'
         - '/volume2/docker/transmition/downloads:/data'
        environment:
            - CREATE_TUN_DEVICE=true
            - OPENVPN_PROVIDER=IVACY
            - OPENVPN_CONFIG=default
            - VPN_CONFIG_SOURCE=default
            - OPENVPN_USERNAME=xxx_user_xxx
            - OPENVPN_PASSWORD=xx_pass_xx
            - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
            - WEBPROXY_ENABLED=false
            - LOCAL_NETWORK=192.168.1.0/16
            - DNS=0.0.0.0,0.0.0.0
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '9091:9091'
            - '9117:9117'
            - '7878:7878'
            - '8989:8989'
        image: haugene/transmission-openvpn

I cant seem to get it to work and now I dont see any errors

the log ->

Starting container with revision: 6922dd6b112d63e099b98165d7cadeaf411b7800

Creating TUN device /dev/net/tun

Using OpenVPN provider: IVACY

Running with VPN_CONFIG_SOURCE default

Starting OpenVPN using config default.ovpn

Modifying /etc/openvpn/ivacy/default.ovpn for best behaviour in this container

Modification: Point auth-user-pass option to the username/password file

sed: cannot rename /etc/openvpn/ivacy/sedTCZdVC: Device or resource busy

Modification: Change ca certificate path

sed: cannot rename /etc/openvpn/ivacy/sedTcJ2QO: Device or resource busy

Modification: Change ping options

sed: cannot rename /etc/openvpn/ivacy/sedb1bOsP: Device or resource busy

sed: cannot rename /etc/openvpn/ivacy/sedWpGf6T: Device or resource busy

sed: cannot rename /etc/openvpn/ivacy/sedISirK0: Device or resource busy

sed: cannot rename /etc/openvpn/ivacy/sedXFgbX5: Device or resource busy

Modification: Update/set resolv-retry to 15 seconds

sed: cannot rename /etc/openvpn/ivacy/sedpyHGt9: Device or resource busy

sed: cannot rename /etc/openvpn/ivacy/sedoxAQgd: Device or resource busy

Modification: Change tls-crypt keyfile path

sed: cannot rename /etc/openvpn/ivacy/sedmvZ9wq: Device or resource busy

Modification: Set output verbosity to 3

sed: cannot rename /etc/openvpn/ivacy/sedHpzaZu: Device or resource busy

sed: cannot rename /etc/openvpn/ivacy/sedQWfb5z: Device or resource busy

Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop

sed: cannot rename /etc/openvpn/ivacy/sedUypXeE: Device or resource busy

sed: cannot rename /etc/openvpn/ivacy/sedfurNLI: Device or resource busy

Setting OpenVPN credentials...

adding route to local network 192.168.1.0/16 via 172.17.0.1 dev eth0

RTNETLINK answers: Invalid argument

Wed Oct 20 21:36:46 2021 WARNING: file '/etc/openvpn/ivacy/Wdc.key' is group or others accessible

Wed Oct 20 21:36:46 2021 WARNING: file '/etc/openvpn/ivacy/openvpn-credentials.txt' is group or others accessible

Wed Oct 20 21:36:46 2021 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021

Wed Oct 20 21:36:46 2021 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10

Wed Oct 20 21:36:46 2021 WARNING: No server certificate verification method has been enabled.  See http://openvpn.net/howto.html#mitm for more info.

Wed Oct 20 21:36:46 2021 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts

Wed Oct 20 21:36:46 2021 Outgoing Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication

Wed Oct 20 21:36:46 2021 Incoming Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication

Wed Oct 20 21:36:46 2021 TCP/UDP: Preserving recently used remote address: [AF_INET]172.94.87.132:80

Wed Oct 20 21:36:46 2021 Socket Buffers: R=[212992->212992] S=[212992->212992]

Wed Oct 20 21:36:46 2021 UDP link local (bound): [AF_INET][undef]:1194

Wed Oct 20 21:36:46 2021 UDP link remote: [AF_INET]172.94.87.132:80

Wed Oct 20 21:37:46 2021 [UNDEF] Inactivity timeout (--ping-exit), exiting

Wed Oct 20 21:37:46 2021 SIGTERM[soft,ping-exit] received, process exiting

**Comment by ThomasBPG** [comment, marked as answer]
Hey, the Ivacy OVPN configurations in https://github.com/haugene/vpn-configs-contrib/tree/main/openvpn/ivacy is pretty messed up and outdated. I've created a pull request in https://github.com/haugene/vpn-configs-contrib/pull/77 and when that goes through, it will be a lot easier to use the container with Ivacy.

On your specific problem: You are connecting to the TCP connection point in Panama using UDP. Either use the right URL from https://support.ivacy.com/servers-list/  or change the protocol - not sure that's the cause of the sed-errors you get, but start here.

(or await my pull request and it should work fine with the standard container configuration for Ivacy and Panama"

**Reply by pkishino** [reply]
Merged

**Reply by caladol1** [reply]
thanks  the original filed did not work so I edit stuff aroun, cant belive I missed that.

proto udp -> proto tdp 
works!!

**Reply by Markiempje525** [reply]
I know this topic is somewhat old. kind of helped me debug some things. Just wanted to shoutout ivacy has just sent me new ovpn files that solved my issue of not being able to connect. Is there a place for me to upload these files? (first post on github :O)

**Reply by ThomasBPG** [reply]
Hey, if you have the updated OVPN files somewhere where I can download them, I'll be happy to make a pull request with the updates like I did above.

**Reply by Markiempje525** [reply]
https://we.tl/t-92UBQ3od3M
This is how it ended up working, for others that use Ivacy...

sudo docker run --cap-add=NET_ADMIN -d \
		-v ---your-storage-folder---:/data \
		-v /transmission-openvpn/config/:/config \
		-e OPENVPN_PROVIDER=ivacy \
 	        -e OPENVPN_CONFIG="NL_Amsterdam_TCP" \   
		-e OPENVPN_USERNAME=---your-accountname---\   % the account name is not your email! 
		-e OPENVPN_PASSWORD=---your-password--- \
		-e LOCAL_NETWORK=192.168.1.0/24 \
		-e “TRANSMISSION_DOWNLOAD_DIR=/data/downloads” \
              --log-driver json-file \
              --log-opt max-size=10m \
              -p 9091:9091 \
              haugene/transmission-openvpn

**Reply by Ferk** [reply]
> https://we.tl/t-92UBQ3od3M

The link expired.
Are the files any different from the ones in  https://ivacy.s3.amazonaws.com/support/OpenVPN-Configs-with-certificate.rar ?
Those are dated 2023.. they were working fine until recently
